### PR TITLE
Ferigis.remove harcoded localhost

### PR DESCRIPTION
**Pull request by ferigis**
Removing harcoded references to `localhost`

### CODE REVIEW

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 1]
```diff
@@ -139,21 +139,21 @@ validation_property(_) ->
...
     ?ae(calendar:gregorian_seconds_to_datetime(ExpectedSeconds),
         ActualDT).
 
 choose_key_by_token_type(_) ->
     %% given mocked keystore (see init_per_testcase)
-    JID = jid:from_binary(<<"alice@localhost">>),
+    JID = jid:from_binary(<<"alice@", domain()/binary>>),
     %% when mod_auth_token asks for key for given token type
     %% then the correct key is returned
     ?ae(<<"access_or_refresh">>, ?TESTED:get_key_for_user(access, JID)),
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Don't use macros (like TESTED on line 159) as module names.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 2]
```diff
@@ -139,21 +139,21 @@ validation_property(_) ->
...
-    ok = ?TESTED:start(<<"localhost">>,
-                       validity_period_cfg(access, {13, hours})),
+    ok = ?TESTED:start(domain(), validity_period_cfg(access, {13, hours})),
     UTCSeconds = utc_now_as_seconds(),
     ExpectedSeconds = UTCSeconds + (    13 %% hours
                                     * 3600 %% seconds per hour
                                    ),
     %% when
-    ActualDT = ?TESTED:expiry_datetime(<<"localhost">>, access, UTCSeconds),
+    ActualDT = ?TESTED:expiry_datetime(domain(),
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Don't use macros (like TESTED on line 148) as module names.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 3]
```diff
@@ -139,21 +139,21 @@ validation_property(_) ->
 
 validity_period_test(_) ->
     %% given
-    ok = ?TESTED:start(<<"localhost">>,
-                       validity_period_cfg(access, {13, hours})),
+    ok = ?TESTED:start(domain(), validity_period_cfg(access, {13, hours})),
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Don't use macros (like TESTED on line 142) as module names.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 4]
```diff
@@ -197,7 +197,7 @@ revoked_token_is_not_valid(_) ->
     self() ! {valid_seq_no, ValidSeqNo},
     T = #token{type = refresh,
                expiry_datetime = ?TESTED:seconds_to_datetime(utc_now_as_seconds() + 10),
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Don't use macros (like TESTED on line 199) as module names.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 5]
```diff
@@ -272,62 +272,67 @@ mock_ejabberd_commands() ->
...
+         {xmlel, <<"lnnitxm">>,
+          [{<<"qytehi">>, <<"axl">>},
+           {<<"xaxforb">>, <<"jrdeydsqhj">>}],
           []},
-         {xmlcdata,<<"pncgsaxl">>},
-         {xmlel,<<"jfofazuau">>,[{<<"si">>,<<"l">>}],[]}]},
-       {xmlel,<<"moy">>,
-        [{<<"femjc">>,<<"qqb">>},{<<"tirfmekvpk">>,<<"sa">>}],
+         {xmlcdata, <<"pncgsaxl">>},
+         {xmlel, <<"jfofazuau">>, [{<<"si">>,<<"l">>}], []}]},
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Missing space right "," on line 294

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 6]
```diff
@@ -197,7 +197,7 @@ revoked_token_is_not_valid(_) ->
     self() ! {valid_seq_no, ValidSeqNo},
     T = #token{type = refresh,
                expiry_datetime = ?TESTED:seconds_to_datetime(utc_now_as_seconds() + 10),
-               user_jid = jid:from_binary(<<"alice@localhost">>),
+               user_jid = jid:from_binary(<<"alice@", domain()/binary>>),
                sequence_no = RevokedSeqNo},
     Revoked = ?TESTED:serialize(?TESTED:token_with_mac(T)),
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Don't use macros (like TESTED on line 202) as module names.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 7]
```diff
@@ -272,62 +272,67 @@ mock_ejabberd_commands() ->
 
 provision_token_example() ->
     {token,provision,
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Missing space right "," on line 274

File: apps/ejabberd/test/commands_backend_SUITE.erl [hunk 8]
```diff
@@ -377,7 +376,7 @@ post_simple_client(_Config) ->
     Arg1 = {title, <<"Juliet's despair">>},
     Arg2 = {content, <<"If they do see thee, they will murder thee!">>},
     Base = <<"/api/ohmyromeo">>,
-    Username = <<"username@localhost">>,
+    Username = <<"username@", domain()/binary>>,
     Auth = {binary_to_list(Username), "secret"},
     Result = binary_to_list(post_simple_client_command(Username, element(2, Arg1), element(2, Arg2))),
```
Comment (gadgetci) [inline apps/ejabberd/test/commands_backend_SUITE.erl]: According to **Elvis**:
> Line 381 is too long:     Result = binary_to_list(post_simple_client_command(Username, element(2, Arg1), element(2, Arg2))),.

File: apps/ejabberd/test/commands_backend_SUITE.erl [hunk 9]
```diff
@@ -8,7 +8,6 @@
 -include_lib("ejabberd/include/ejabberd_commands.hrl").
 
 -define(PORT, 5288).
--define(HOST, "localhost").
 -define(IP,  {127,0,0,1}).
```
Comment (gadgetci) [inline apps/ejabberd/test/commands_backend_SUITE.erl]: According to **Elvis**:
> Missing space right "," on line 11

Comment (gadgetci) [inline apps/ejabberd/test/commands_backend_SUITE.erl]: According to **Elvis**:
> Missing space right "," on line 11

Comment (gadgetci) [inline apps/ejabberd/test/commands_backend_SUITE.erl]: According to **Elvis**:
> Missing space right "," on line 11

File: apps/ejabberd/test/commands_backend_SUITE.erl [hunk 10]
```diff
@@ -190,7 +189,7 @@ post_simple_with_subcategory(_Config) ->
     check_location_header(Response, list_to_binary(build_path_prefix()++"/api/weather/10/subcategory/" ++ Result)).
```
Comment (gadgetci) [inline apps/ejabberd/test/commands_backend_SUITE.erl]: According to **Elvis**:
> Missing space left "++" on line 189

Comment (gadgetci) [inline apps/ejabberd/test/commands_backend_SUITE.erl]: According to **Elvis**:
> Line 189 is too long:     check_location_header(Response, list_to_binary(build_path_prefix()++"/api/weather/10/subcategory/" ++ Result))..

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 11]
```diff
@@ -127,38 +126,42 @@ validation_test(Config) ->
...
+    Domain = domain(),
+    JID = jid:from_binary(<<"alice@", Domain/binary>>),
+    Tested = tested(),
     %% when mod_auth_token asks for key for given token type
     %% then the correct key is returned
-    ?ae(<<"access_or_refresh">>, ?TESTED:get_key_for_user(access, JID)),
-    ?ae(<<"access_or_refresh">>, ?TESTED:get_key_for_user(refresh, JID)),
-    ?ae(<<"provision">>, ?TESTED:get_key_for_user(provision, JID)).
+    ?AE(<<"access_or_refresh">>, Tested:get_key_for_user(access, JID)),
+    ?AE(<<"access_or_refresh">>, Tested:get_key_for_user(refresh, JID)),
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 163. Only modules that define callbacks should make dynamic calls.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 12]
```diff
@@ -173,11 +176,13 @@ is_join_and_split_with_base16_and_zeros_reversible(RawToken) ->
     end.
 
 is_serialization_reversible(Token) ->
-    Token =:= ?TESTED:deserialize(?TESTED:serialize(Token)).
+    Tested = tested(),
+    Token =:= Tested:deserialize(Tested:serialize(Token)).
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 180. Only modules that define callbacks should make dynamic calls.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 13]
```diff
@@ -357,24 +369,25 @@ make_token({Type, Expiry, JID, SeqNo, VCard}) ->
...
                expiry_datetime = Expiry,
                user_jid = jid:from_binary(JID)},
+    Tested = tested(),
     case Type of
         access ->
-            ?TESTED:token_with_mac(T);
+            Tested:token_with_mac(T);
         refresh ->
-            ?TESTED:token_with_mac(T#token{sequence_no = SeqNo});
+            Tested:token_with_mac(T#token{sequence_no = SeqNo});
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 377. Only modules that define callbacks should make dynamic calls.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 14]
```diff
@@ -357,24 +369,25 @@ make_token({Type, Expiry, JID, SeqNo, VCard}) ->
     T = #token{type = Type,
                expiry_datetime = Expiry,
                user_jid = jid:from_binary(JID)},
+    Tested = tested(),
     case Type of
         access ->
-            ?TESTED:token_with_mac(T);
+            Tested:token_with_mac(T);
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 375. Only modules that define callbacks should make dynamic calls.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 15]
```diff
@@ -127,38 +126,42 @@ validation_test(Config) ->
...
 validation_property(_) ->
     prop(validation_property,
          ?FORALL(Token, valid_token(), is_valid_token_prop(Token))).
 
 validity_period_test(_) ->
     %% given
-    ok = ?TESTED:start(<<"localhost">>,
-                       validity_period_cfg(access, {13, hours})),
+    Tested = tested(),
+    ok = Tested:start(domain(), validity_period_cfg(access, {13, hours})),
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 143. Only modules that define callbacks should make dynamic calls.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 16]
```diff
@@ -194,14 +199,16 @@ revoked_token_is_not_valid(_) ->
...
+    Tested = tested(),
     self() ! {valid_seq_no, ValidSeqNo},
     T = #token{type = refresh,
-               expiry_datetime = ?TESTED:seconds_to_datetime(utc_now_as_seconds() + 10),
-               user_jid = jid:from_binary(<<"alice@localhost">>),
+               expiry_datetime = Tested:seconds_to_datetime(utc_now_as_seconds() + 10),
+               user_jid = jid:from_binary(<<"alice@", Domain/binary>>),
                sequence_no = RevokedSeqNo},
-    Revoked = ?TESTED:serialize(?TESTED:token_with_mac(T)),
+    Revoked = Tested:serialize(Tested:token_with_mac(T)),
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 209. Only modules that define callbacks should make dynamic calls.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 17]
```diff
@@ -127,38 +126,42 @@ validation_test(Config) ->
...
-    JID = jid:from_binary(<<"alice@localhost">>),
+    Domain = domain(),
+    JID = jid:from_binary(<<"alice@", Domain/binary>>),
+    Tested = tested(),
     %% when mod_auth_token asks for key for given token type
     %% then the correct key is returned
-    ?ae(<<"access_or_refresh">>, ?TESTED:get_key_for_user(access, JID)),
-    ?ae(<<"access_or_refresh">>, ?TESTED:get_key_for_user(refresh, JID)),
-    ?ae(<<"provision">>, ?TESTED:get_key_for_user(provision, JID)).
+    ?AE(<<"access_or_refresh">>, Tested:get_key_for_user(access, JID)),
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 162. Only modules that define callbacks should make dynamic calls.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 18]
```diff
@@ -357,24 +369,25 @@ make_token({Type, Expiry, JID, SeqNo, VCard}) ->
...
     case Type of
         access ->
-            ?TESTED:token_with_mac(T);
+            Tested:token_with_mac(T);
         refresh ->
-            ?TESTED:token_with_mac(T#token{sequence_no = SeqNo});
+            Tested:token_with_mac(T#token{sequence_no = SeqNo});
         provision ->
-            ?TESTED:token_with_mac(T#token{vcard = VCard})
+            Tested:token_with_mac(T#token{vcard = VCard})
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 379. Only modules that define callbacks should make dynamic calls.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 19]
```diff
@@ -127,38 +126,42 @@ validation_test(Config) ->
...
-                       validity_period_cfg(access, {13, hours})),
+    Tested = tested(),
+    ok = Tested:start(domain(), validity_period_cfg(access, {13, hours})),
     UTCSeconds = utc_now_as_seconds(),
     ExpectedSeconds = UTCSeconds + (    13 %% hours
                                     * 3600 %% seconds per hour
                                    ),
     %% when
-    ActualDT = ?TESTED:expiry_datetime(<<"localhost">>, access, UTCSeconds),
+    ActualDT = Tested:expiry_datetime(domain(),
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 149. Only modules that define callbacks should make dynamic calls.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 20]
```diff
@@ -127,38 +126,42 @@ validation_test(Config) ->
 
 validation_test(_, ExampleToken) ->
     %% given
-    Serialized = ?TESTED:serialize(ExampleToken),
+    Tested = tested(),
+    Serialized = Tested:serialize(ExampleToken),
     %% when
-    Result = ?TESTED:authenticate(Serialized),
+    Result = Tested:authenticate(Serialized),
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 132. Only modules that define callbacks should make dynamic calls.

File: apps/ejabberd/test/auth_tokens_SUITE.erl [hunk 21]
```diff
@@ -127,38 +126,42 @@ validation_test(Config) ->
 
 validation_test(_, ExampleToken) ->
     %% given
-    Serialized = ?TESTED:serialize(ExampleToken),
+    Tested = tested(),
+    Serialized = Tested:serialize(ExampleToken),
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_tokens_SUITE.erl]: According to **Elvis**:
> Remove the dynamic function call on line 130. Only modules that define callbacks should make dynamic calls.

**Comment by michalwski** [comment]
Good job with fixing all the Elvis's comments! Could you try rearrange commits? The ideal situation would be 2 commits - one for removing hardcoded domain, the second for fixing all the Elvis's comments.

**Comment by michalwski** [comment]
Could you do the same kind of refactoring in big tests (`test/ejabberd_tests`). This should be done in a separate PR, this one is big enough :)

**Comment by ferigis** [comment]
sure, actually apps/ejabberd/tests is not completed yet, when I complete it I will fix the commits

**Comment by michalwski** [comment]
I'm closing this PR as it's not that important in small tests. Similar change was done and merged for big tests.

**Pull request closed without merging**